### PR TITLE
Fix the DevShop Control composer template

### PR DESCRIPTION
**Pull request by jonpugh**
- Preserve paths is not needed for installing `devshop/devmaster`. It is a real package.
- We should check that it gets installed properly regardless if the repository is local or packagist.
- First run of composer install does not actually install the ./web/profiles/devmaster package.

### CODE REVIEW

File: src/DevShop/Component/DevShopControlTemplate/composer.json [hunk 1]
```diff
@@ -145,9 +143,8 @@
             "web/sites/all/modules/contrib/{$name}/": ["type:drupal-module"],
             "web/sites/all/themes/contrib/{$name}/": ["type:drupal-theme"]
       },
-        "preserve-paths": [
-            "web/sites",
-            "web/profiles"
```
Comment (matthandus) [inline src/DevShop/Component/DevShopControlTemplate/composer.json]: Did you mean to delete this line 150 to preserve "web/profiles"? There's a fail in the auto test that could be related. Error message says missing the devshop.info file.

File: src/DevShop/Component/DevShopControlTemplate/scripts/composer/ScriptHandler.php [hunk 2]
```diff
@@ -14,6 +14,24 @@
...
+    $fs = new Filesystem();
+    $io = $event->getIO();
+    $drupalFinder = new DrupalFinder();
+    $drupalFinder->locateRoot(getcwd());
+    $drupalRoot = $drupalFinder->getDrupalRoot();
+
+    $devmaster_path = $drupalRoot . '/profiles/devmaster';
+    $devmaster_info_path = $devmaster_path  . '/devmaster.info';
+    if (!$fs->exists($devmaster_info_path)) {
+      $io->writeError('<error>Devmaster profile info file was not found at ' . $devmaster_info_path . '</error>');
```
Comment (matthandus) [inline src/DevShop/Component/DevShopControlTemplate/scripts/composer/ScriptHandler.php]: Ah this is the error message I saw in the test that failed. 
`if (!$fs->exists($devmaster_info_path))` returns TRUE in the automatic test.

**Review by matthandus — CHANGES_REQUESTED** [review]
Looks like the devshop.info is not found causing this to fail test. Is that related to deleting line 150 to preserve "web/profiles"?

File: src/DevShop/Component/DevShopControlTemplate/scripts/composer/ScriptHandler.php (hunk 2 above)
Comment (matthandus) [reply src/DevShop/Component/DevShopControlTemplate/scripts/composer/ScriptHandler.php]: Here's the current string that should be the path to the .info file.
`/usr/share/devshop/src/DevShop/Component/DevShopControlTemplate/web/profiles/devmaster/devmaster.info`

Comment (jonpugh) [reply src/DevShop/Component/DevShopControlTemplate/scripts/composer/ScriptHandler.php]: Removing the preserve path fixes it from installing directly.

Now it's failing when using the custom composer "repository" that installs via symlink.

Whipping up code to detect the symlink and try to read the full path.

The path it's looking for is in https://github.com/opendevshop/devshop/pull/632/checks?check_run_id=1449549897#step:4:648

`/usr/share/devshop/src/DevShop/Component/DevShopControlTemplate/web/profiles/devmaster/devmaster.info`

**Comment by jonpugh** [comment]
Ok it looks close.

Everything passes except for Test DevShop, which seems to lose the symlink to devmaster.

I was unable to reproduce this locally, even with the path repo, so we should probably start there.

**Comment by jonpugh** [comment]
The test that is failing is the devshop composer template project install.

It's possible this is because of composer2 vs composer1.

We need to import the devshop-composer-template project into the devshop megarepo so that we can properly test.

https://github.com/opendevshop/devshop/pull/632/checks?check_run_id=1455805693#step:7:649

**Comment by jonpugh** [comment]
Everything here was brought into other PRs. This can be closed since all that's left in the diff is a hack!

**Pull request closed without merging**